### PR TITLE
[civ2][data/1] move arrow 12 data tests to civ2

### PR DESCRIPTION
**Pull request by can-anyscale**
Move arrow 12 data tests to civ2
- Need to fix one test that hard code /ray as the mount name dir
- Add --test-env and --build-name flag to ray_ci, since data requires custom build name and test environment

Test:
- CI

### CODE REVIEW

File: ci/docker/base.ml.Dockerfile
Comment (can-anyscale) [inline ci/docker/base.ml.Dockerfile:6]: this part is required to have nvm

File: python/ray/air/BUILD
Comment (can-anyscale) [inline python/ray/air/BUILD:80]: for data team, can we also remove ml as owner?

File: python/ray/data/tests/test_image.py
Comment (can-anyscale) [inline python/ray/data/tests/test_image.py:67]: this is required since the mount dir in civ2 is rayci

File: ci/docker/base.ml.Dockerfile
Comment (aslonnie) [reply ci/docker/base.ml.Dockerfile:6]: this file is from civ1. why this is required?

can nvm be installed via some other way in `data.build.Dockerfile`?

Comment (can-anyscale) [reply ci/docker/base.ml.Dockerfile:6]: it can yes, but we'll need to do the same for for ml.build.Dockerfile and rllib.build.Dockerfile

in civ1, nvm is installed twice, once in base.build.Dockerfile and once is build.Dockerfile; and the ml image used for data/ml/rllib is built through a layer of base.test < base.ml < build < ml which has a whole lot of useless stuff (if you look at base.ml and ml they are exactly the same)

civ2 ignore build.Dockerfile (which basically builds ray) and attempt to build ray as part of rayci itself; which requires the nvm to be installed; but putting nvm into wanda is nicer for caching so i just move it to the base itself.

File: python/ray/data/tests/test_image.py
Comment (aslonnie) [reply python/ray/data/tests/test_image.py:67]: feels that this should be something that needs to be fixed in the test itself?

like assuming the abs path to run the test means that bazel test with this test will fail locally?

like will this fail the tests on other arrow versions on civ1?

**Review by aslonnie — COMMENTED** [review]
very happy to see data tests passing on civ2!

guess we still need approvals from the code owners.

**Comment by aslonnie** [comment]
unless we are also migrating other flavors of data tests together, we would need to maintain civ1 to be passing.

like all these are failing now on:

https://buildkite.com/ray-project/oss-ci-build-pr/builds/34197#018a40c8-d1ad-4ad1-83d7-273b71b437d4/1922-2180

**Comment by can-anyscale** [comment]
Lolll you're right, now that particular test test will break in civ1

**Comment by can-anyscale** [comment]
Actually fix them in subsequent PR https://github.com/ray-project/ray/pull/39044 w00h00

File: python/ray/air/BUILD
Comment (can-anyscale) [reply python/ray/air/BUILD:80]: CC: @c21, @raulchen

**Comment by can-anyscale** [comment]
I'll merge this together with https://github.com/ray-project/ray/pull/39044 so that there are so straggler failing tests on civ1

File: python/ray/data/tests/test_image.py
Comment (can-anyscale) [reply python/ray/data/tests/test_image.py:67]: I'll leave this for the data team to improve

**Review by aslonnie — APPROVED** [review]
content lgtm.

but please don't merge this. maybe just merge the downstream into master altogether.

**Comment by can-anyscale** [comment]
@aslonnie: totally, i'll merge them together. Also this PR depends on this PR https://github.com/ray-project/ray/pull/38737 that needs some review ;))

This PR needs approval from a few code owners, I guess mostly from data team, CC: @c21 or @amogkam , thankkks

**Comment by can-anyscale** [comment]
@scottjlee this PR will bring sharding to runtime and reduce your time-to-signal wait time from 1 hour -> 30 minutes ;)

File: python/ray/air/BUILD
Comment (scottjlee) [reply python/ray/air/BUILD:72]: looking at test_data_batch_conversion.py, it looks to be testing mostly AIR utility methods, and not Ray Data directly. IMO we should not add this under data team?

File: python/ray/data/tests/test_image.py
Comment (scottjlee) [reply python/ray/data/tests/test_image.py:67]: i think we can generalize this by verifying the suffix of the string, without fear of losing correctness:
```
assert (
    len(ds.input_files()) == 1 and 
    ds.input_files()[0].endswith("ray/data/examples/data/image-datasets/simple/image1.jpg")
)
```

**Review by scottjlee — APPROVED** [review]
Data tests changes LGTM

File: .buildkite/data.rayci.yml
Comment (scottjlee) [inline .buildkite/data.rayci.yml:3]: nit: Arrow 12 tests (also same with the flakey test label)

File: .buildkite/pipeline.ml.yml
Comment (scottjlee) [inline .buildkite/pipeline.ml.yml:416]: just to confirm, is the manual installation for `datasets==2.14.0` no long needed because it's handled gracefully by the new CI workflow?

File: python/ray/data/tests/test_image.py
Comment (can-anyscale) [reply python/ray/data/tests/test_image.py:67]: I'll update the code with this

File: python/ray/air/BUILD
Comment (can-anyscale) [reply python/ray/air/BUILD:72]: got you, I'll keep team:ml as the first oncall then, and just leave team:data there for now so that it's compatible with the ray civ2 interprets tests

I'll remove the team:data part once I migrate ml tests to civ2

**Pull request merged**
